.github/workflows/azure_e2e_test.yml

```diff
@@ -78,51 +60,44 @@ jobs:
+          inputs: >-
+            {
               "bee-name": "${{ env.BEE_NAME }}",
               "billing-project": "${{ needs.params-gen.outputs.project-name }}",
-              "billing-project-creator": "${{ inputs.owner-subject }}",
-              "service-account": "${{inputs.service-account}}" }'
+              "service-account": "firecloud-qa@broad-dsde-qa.iam.gserviceaccount.com"
+            }
 
-  run-workflows-azure-e2e-test:
-    runs-on: ubuntu-latest
+  run-cbas-azure-e2e-test:
     needs: [ params-gen, create-and-attach-billing-project-to-landing-zone-workflow ]
+    permissions:
+      contents: read
+      id-token: write
+    uses: broadinstitute/dsp-reusable-workflows/.github/workflows/run_cbas_azure_e2e_tests.yaml@main
+    with:
+      bee-name: '${{ needs.params-gen.outputs.bee-name }}'
+      billing-project-name: '${{ needs.params-gen.outputs.project-name }}'
+
+  destroy-bee-workflow:
+    runs-on: ubuntu-latest
+    needs: [create-bee-workflow, run-cbas-azure-e2e-test]
+    if: always()
     permissions:
       contents: 'read'
       id-token: 'write'
     steps:
-      - uses: 'actions/checkout@v3'
-      - name: Generate OAuth2 2.0 access token for owner
-        id: 'owner_auth'
-        uses: google-github-actions/auth@v1
-        with:
-          token_format: 'access_token'
-          workload_identity_provider: 'projects/1038484894585/locations/global/workloadIdentityPools/github-wi-pool/providers/github-wi-provider'
-          service_account: ${{ inputs.service-account }}
-          access_token_scopes: 'profile, email, openid'
-          access_token_subject: ${{ inputs.owner-subject }}
-          export_environment_variables: false
-          create_credentials_file: false
-      - uses: actions/checkout@v3
-        with:
-          #NOTE: replace with ${{inputs.target-branch}} before merge
-          #TODO: this might change of the python script is moved to dsp-reusable-workflows
-          ref: refs/heads/sps_azure_e2e_test
-      - name: Set up python
-        id: setup-python
-        uses: actions/setup-python@v4
+      - name: dispatch to terra-github-workflows
+        uses: broadinstitute/workflow-dispatch@v3
         with:
-          python-version: '3.11'
-      - name: Install poetry
-        uses: snok/install-poetry@v1
-      - name: Install dependencies
-        run: |
-          poetry install --no-interaction
-      - name: Run e2e test
-        env:
-          BEE_NAME: ${{env.BEE_NAME}}
-          BILLING_PROJECT_NAME: ${{needs.params-gen.outputs.project-name}}
-          BEARER_TOKEN: ${{ steps.owner_auth.outputs.access_token }}
-        run: |
-          poetry run e2e-test/azure_e2e_test.py
+          workflow: bee-destroy
+          repo: broadinstitute/terra-github-workflows
+          ref: refs/heads/main
+          wait-for-completion: false
+          token: '${{ env.BROADBOT_TOKEN }}'
+          inputs: '{ "bee-name": "${{ env.BEE_NAME }}" }'
+
+  report-workflow:
```

Comment:
Does this also (like `delete_bee`) need some `needs` and `if` parameters to make sure it always runs, and runs _after_ everything else?

Reply:
Based on [your example](https://github.com/DataBiosphere/cbas/actions/runs/6597974881), it looks like the report happened while the status was still `queued`

Reply:
Based on my understanding this runs separately from the test run pipeline and runs in parallel to the `params-gen` job. So this step is always run even before the test starts. My understanding is that the step basically tells sherlock that monitor this new workflow run and send Slack notification when the run fails. 
https://github.com/broadinstitute/sherlock/blob/fd3ad8406edd0223713fb59a7124354125a91a1b/.github/workflows/client-report-workflow.yaml#L30-L33

Reply:
oh I see, ok 👍